packages/css/src/utils.ts

```diff
@@ -0,0 +1,15 @@
+import type { ColorModesScale, Theme, ThemeStyles } from './types'
+
+/**
+ * Constrained identity function used to constrain user's theme type to Theme
+ * while preserving its exact type.
+ */
+export const makeTheme = <T extends Theme>(theme: T): T => theme
```

Comment:
This looks awesome, is it internal API or intended to be used from the outside?

(Wondering if consumers could use it to build custom themes with or without presets?)

Reply:
I try to prefix exported internal APIs with `__` or even `__internal` and mark add `@internal` jsdoc comment. This one is user-facing. I've been defining it in my projects for a long time.